pallets/lottery/src/lib.rs

```diff
@@ -0,0 +1,351 @@
+//! # Lottery Monetization Module
+//!
+//! The Monetization module by lottery method
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Decode, Encode};
+use frame_support::{
+	decl_error, decl_event, decl_module, decl_storage,
+	dispatch::DispatchResult,
+	ensure,
+	traits::{Currency},
+};
+use frame_system::{self as system, ensure_signed};
+use log;
+use sp_runtime::{
+	sp_std::convert::TryInto,
+	traits::{One, Zero},
+	RuntimeDebug,
+};
+use sp_std::{cmp::Ordering, collections::btree_map::BTreeMap, ops::Add, prelude::*};
+
+use pallet_posts::Post;
+use pallet_utils::PostId;
+
+// TODO: move all tests to df-integration-tests
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as system::Config>::AccountId>>::Balance;
+/// The pallet's configuration trait.
+pub trait Config: system::Config + pallet_posts::Config {
+	/// The overarching event type.
+	type Event: From<Event<Self>> + Into<<Self as system::Config>::Event>;
+
+	type Currency: Currency<Self::AccountId>;
```

Comment:
You can re-use this from `pallet_utils::Config`, so you future usages will be like `<T as pallet_utils::Config>::Currency...`